testutil/testnode/full_node.go

```diff
@@ -195,13 +200,17 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (cleanup func() error
 	tmNode, app, cctx, err := New(t, DefaultParams(), tmCfg, false, genState, kr)
 	require.NoError(t, err)
 
+	// locking the mutex not to be able to spin up multiple nodes at the same time.
+	mut.Lock()
```

Comment:
```suggestion
	mut.Lock()
	t.Cleanup(mut.Unlock)
```
I would personally do it this way. Perhaps even remove the custom cleanup function since the testing fixture already provides it. There are two cases that I can see where the code exits before calling `Unlock`. Also I generally prefer to not pass responsibility for cleanup to a function outside the scope where the resources were created (if possible).

Reply:
> Perhaps even remove the custom cleanup function since the testing fixture already provides it

Understandable, we could also define a certain `TestNode` struct that contains an exported `Cleanup` function to be used in tests. However, I am not against the way it is right now as it is really easy to start and stop.

> There are two cases that I can see where the code exits before calling Unlock

Yes, but those are not returning errors, they're `require`ing no errors. This means that, if I understand right, if we hit that case, the whole test suite will tear down and there will be no need to unlock nor cleanup.

This brings a question to my mind, will starting a separate test in a separate package, after failing a require, be affected by the locked mutex? I assume not, but not sure.

Reply:
if we can stick everything in the testing.T.Cleanup, then we totally should. @SweeXordious if you think this is out of scope of this PR, then I can handle it in a follow up

Reply:
I can open a follow up PR doing that. Do we want to  merge this one?